mempool/src/interface/mempool_interface.rs

```diff
@@ -23,17 +23,23 @@ use subsystem::{CallRequest, ShutdownRequest};
 
 pub trait MempoolInterface: Send + Sync {
     /// Add a transaction to mempool
-    fn add_transaction(&mut self, tx: SignedTransaction) -> Result<(), Error>;
+    fn add_transaction(&mut self, tx: SignedTransaction) -> Result<TxStatus, Error>;
```

Comment:
Just to be clear because I read this in the PR description: If a transaction is not validated, we should never forward it. This is probably a big, sensitive problem that we should double-check what other blockchains do to mitigate this issue. 

I understand it's a trade-off between losing transactions and relaying invalid transactions. Relaying invalid transactions should be avoided at all cost, though. This is how I see, unless a better argument is provided.

Reply:
I agree. The comment in the description is concerned with the scenario where a transaction is initially put in orphan pool so not fully validated but later another transaction comes which makes it possible to fully validate the first transaction. We can then forward it since it has been checked for issues (or punish the originating peer if it turns out to be invalid).